### PR TITLE
Make sure $ATTR is filled in with react-http-leak rule

### LINKED ISSUE #1715: react-http-leak doesn't fill $ATTR in most cases
Semgrep reports:

> This HTML element 'img' and attribute '$ATTR' together may load an external resource.

I expect:

> This HTML element 'img' and attribute 'src' together may load an external resource.

[semgrep-rules/react-http-leak.yaml at develop · returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/blob/develop/typescript/react/security/audit/react-http-leak.yaml)

The message in this rule is:

> This HTML element '$EL' and attribute '$ATTR' together may load an external resource.

But the `$ATTR` variable is only used for attributes of `<object>`, so it's not set for `` and many other tags.

### PR DESCRIPTION
**Pull request by minusworld**
Fixes https://github.com/returntocorp/semgrep-rules/issues/1715

### CODE REVIEW

**Comment by colleend** [comment]
@emjin I think the CI `--validate` issues are from a rule that you wrote, can you take a look? 🙏 

UPDATE:

Emma found the issue! Lines 162 and 142 don't have any metavariables that the metavariable regex/patterns are flagging

File: typescript/react/security/audit/react-http-leak.yaml [hunk 1]
```diff
@@ -3,112 +3,170 @@ rules:
...
+              metavariable: $EL
+              regex: video
+          - metavariable-regex:
+              metavariable: $ATTR
+              regex: poster
           - pattern-not: |
               <video poster="..." />
       - patterns:
           - pattern: |
               <command icon=... />
```
Comment (colleend) [inline typescript/react/security/audit/react-http-leak.yaml]: we haven't changed this to $EL and $ATTR!

File: typescript/react/security/audit/react-http-leak.yaml [hunk 2]
```diff
@@ -3,112 +3,170 @@ rules:
...
+              metavariable: $EL
+              regex: html
+          - metavariable-regex:
+              metavariable: $ATTR
+              regex: manifest
           - pattern-not: |
               <html manifest="..." />
       - patterns:
           - pattern: |
               <meta content=... />
```
Comment (colleend) [inline typescript/react/security/audit/react-http-leak.yaml]: haven't changed this either

**Review by colleend — CHANGES_REQUESTED** [review]

**Comment by kurt-r2c** [comment]
@colleend can you recheck this PR for the issue(s) you raised? The force push has messed up the history and the point-in-time commit appears to be missing.

**Review by colleend — APPROVED** [review]
cool, just looked over it again and LGTM!

**Pull request merged**